FinalGPTProj/Client.py

```diff
@@ -0,0 +1,69 @@
+import time
+import requests
+
+class Status:
+    def __init__(self, status, filename, timestamp, explanation):
+        self.status = status
+        self.filename = filename
+        self.timestamp = timestamp
+        self.explanation = explanation
+
+    def is_done(self):
+        return self.status == "done"
+
+class GPTExplainerClient:
+    def __init__(self, base_url):
+        self.base_url = base_url
+
+    def upload(self, file_path):
+        url = f"{self.base_url}/upload"
+        files = {"file": open(file_path, "rb")}
+        response = requests.post(url, files=files)
+
+        if response.status_code == 200:
+            json_data = response.json()
+            return json_data["uid"]
+
+        raise Exception("Upload failed")
+
+    def status(self, uid):

```

Comment:
Use type hints and return types -
https://www.geeksforgeeks.org/duck-typing-in-python/